lib/pmdtester/builders/summary_report_builder.rb

```diff
@@ -2,160 +2,76 @@
-
-      FileUtils.mkdir_p(REPORT_DIR) unless File.directory?(REPORT_DIR)
-      index = File.new(INDEX_PATH, 'w')
-
-      html_report = build_html_report('Summary report')
-      copy_css(REPORT_DIR)
-
-      index.puts html_report
-      index.close
-
-      logger.info 'Built summary report successfully!'
-    end
-
-    def get_branch_details(branch_name)
-      details = PmdBranchDetail.new(branch_name)
-      details.load
-    end
-
-    def build_body(doc)
-      build_branch_details_section(doc)
-      build_projects_section(doc)
-    end
-
-    def build_branch_details_section(doc)
-      doc.div(class: 'section', id: 'branchdetails') do
-        doc.h2 'Branch details:'
-        build_branch_details_table(doc)
+    def write_all_projects(projects, base_details, patch_details)
+      projects.each do |project|
+        process_project(project, "#{REPORT_DIR}/#{project.name}")
       end
-    end
-
-    def build_branch_details_table(doc)
-      doc.table(class: 'bodyTable', border: '0') do
-        build_branch_table_head(doc)
-        build_branch_table_body(doc)
-      end
-    end
+      logger.info 'Built all difference reports successfully!'
 
-    def build_branch_table_head(doc)
-      doc.thead do
-        doc.tr do
-          doc.th 'Item'
-          doc.th 'base'
-          doc.th 'patch'
-        end
-      end
-    end
-
-    def build_branch_table_body(doc)
-      doc.tbody do
-        build_branch_name_table_row(doc)
-        build_branch_table_row(doc, 'branch last commit sha', @base_details.branch_last_sha,
-                               @patch_details.branch_last_sha)
-        build_branch_table_row(doc, 'branch last commit message', @base_details.branch_last_message,
-                               @patch_details.branch_last_message)
-        build_branch_table_row(doc, 'total execution time', @base_details.format_execution_time,
-                               @patch_details.format_execution_time)
-        build_branch_table_row(doc, 'jdk version', @base_details.jdk_version,
-                               @patch_details.jdk_version)
-        build_branch_table_row(doc, 'language', @base_details.language,
-                               @patch_details.language)
-        build_branch_config_table_row(doc)
-      end
-    end
-
-    def build_branch_config_table_row(doc)
-      doc.tr do
-        doc.td(class: 'c') { doc.text 'branch configuration' }
-        base_config_src_path = @base_details.target_branch_config_path
-        copy_branch_config_file(base_config_src_path, BASE_CONFIG_PATH)
-        doc.td(class: 'a') do
-          doc.a(href: './base_config.xml') { doc.text 'base config' }
-        end
-        patch_config_stc_path = @patch_details.target_branch_config_path
-        FileUtils.cp(patch_config_stc_path, PATCH_CONFIG_PATH)
-        doc.td(class: 'b') do
-          doc.a(href: './patch_config.xml') { doc.text 'patch config' }
-        end
-      end
+      FileUtils.mkdir_p(REPORT_DIR)
```

Comment:
The directory is created already when rendering the individual projects reports. So, at this point here, the directory already exists and the mkdir wouldn't be needed. However, it might be, that for some unit tests, we don't have any projects, so it's better to keep it here to make sure, the directory is created, if it doesn't exist yet.